api/agent/drivers/docker/docker_client.go

```diff
@@ -294,6 +294,9 @@ func (d *dockerWrap) AttachToContainerNonBlocking(ctx context.Context, opts dock
 	ctx, span := trace.StartSpan(ctx, "docker_attach_container")
 	defer span.End()
 
+	// WARNING: fragile logic below, especially if Success channel is set in opts.
+	// Current impl in d.docker seems safe to repeat the attach even with
+	// Success channel set.
```

Comment:
the `Stats` has a similar issue. honestly we should just skip retry for this one and let it fail. in general, `create container` is the first one that screws up if docker goes down (or pull). it sounds better to have a tiny % fail here than try to reason about this behavior if we ever hit it (task without logs?)